### PR TITLE
[synapse] Fix many README links

### LINKED ISSUE #17666: Azure.Analytics.Synapse.AccessControl Readme issues
1.
Section [link:](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/synapse/Azure.Analytics.Synapse.AccessControl#install-the-package)
image
Suggestion:Add `[nuget]`: https://www.nuget.org/

2.
Section [link:](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/synapse/Azure.Analytics.Synapse.AccessControl#prerequisites)

image
Suggestion:

1.Add `[azure_sub]`: https://azure.microsoft.com/free/

2.Add hyperlink for `Azure Portal`: https://portal.azure.com/

3.Add `[azure_cli]`: https://docs.microsoft.com/cli/azure

3.
Section [link:](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/synapse/Azure.Analytics.Synapse.AccessControl#authenticate-the-client)
image
Suggestion:

1.Add `[accesscontrol_client_class]`: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/synapse/Azure.Analytics.Synapse.AccessControl/src/Customization/AccessControlClient.cs

2.Add `[azure_identity]`: https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/identity/Azure.Identity

3.Add `[DefaultAzureCredential]`: https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/identity/Azure.Identity#defaultazurecredential

@jongio for notification.

### LINKED ISSUE #17667: Azure.Analytics.Synapse.Artifacts Readme issues
1.
Section [link:](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/synapse/Azure.Analytics.Synapse.Artifacts#install-the-package)
image
Suggestion: Add `[nuget]`: https://www.nuget.org/

2.
Section [link:](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/synapse/Azure.Analytics.Synapse.Artifacts#prerequisites)
image
Suggestion:

1.Add `[azure_sub]`: https://azure.microsoft.com/free/

2.Add hyperlink for `Azure Portal`: https://portal.azure.com/

3.Add `[azure_cli]`: https://docs.microsoft.com/cli/azure

3.
Section [link:](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/synapse/Azure.Analytics.Synapse.Artifacts#authenticate-the-client)
image
Suggestion:

1.Add `[development_client_class]` a link;

2.Add `[azure_identity]`: https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/identity/Azure.Identity

3.Add `[DefaultAzureCredential]`: https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/identity/Azure.Identity#defaultazurecredential

@jongio for notification.

### LINKED ISSUE #17668: Azure.Analytics.Synapse.ManagedPrivateEndpoints Readme issues
1.
Section [link:](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/synapse/Azure.Analytics.Synapse.ManagedPrivateEndpoints#install-the-package)
image
Suggestion:Add `[nuget]`: https://www.nuget.org/

2.
Section [link:](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/synapse/Azure.Analytics.Synapse.ManagedPrivateEndpoints#prerequisites)
image
Suggestion:

1.Add `[azure_sub]`: https://azure.microsoft.com/free/

2.Add hyperlink for `Azure Portal`: https://portal.azure.com/

3.Add `[azure_cli]`: https://docs.microsoft.com/cli/azure

3.
Section [link:](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/synapse/Azure.Analytics.Synapse.ManagedPrivateEndpoints#authenticate-the-client)
image
Suggestion:

1.Update  `[ManagedPrivateEndpointClient][managed_priate_endpoint_client_class] ` to  `[ManagedPrivateEndpointsClient][managed_priate_endpoints_client_class]`
And add `[managed_priate_endpoints_client_class]`: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/synapse/Azure.Analytics.Synapse.ManagedPrivateEndpoints/src/Customization/ManagedPrivateEndpointsClient.cs

2.Add `[azure_identity]`: https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/identity/Azure.Identity

3.Add `[DefaultAzureCredential]`: https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/identity/Azure.Identity#defaultazurecredential

4.
Section [link:](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/synapse/Azure.Analytics.Synapse.ManagedPrivateEndpoints#key-concepts)
image
Suggestion: Update  `ManagedPrivateEndpointClient`  to `ManagedPrivateEndpointsClient`

@jongio for notification.

### LINKED ISSUE #17669: Azure.Analytics.Synapse.Monitoring Readme issues
1.
Section [link:](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/synapse/Azure.Analytics.Synapse.Monitoring#install-the-package)
image
Suggestion:Add `[nuget]`: https://www.nuget.org/

2.
Section [link:](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/synapse/Azure.Analytics.Synapse.Monitoring#prerequisites)
image
Suggestion:

1.Add `[azure_sub]`: https://azure.microsoft.com/free/

2.Add hyperlink for `Azure Portal:` https://portal.azure.com/

3.Add `[azure_cli]`: https://docs.microsoft.com/cli/azure

3.
Section [link:](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/synapse/Azure.Analytics.Synapse.Monitoring#authenticate-the-client)
image
Suggestion:

1.Add `[monitoring_client_class]`: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/synapse/Azure.Analytics.Synapse.Monitoring/src/Customization/MonitoringClient.cs

2.Add `[azure_identity]`: https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/identity/Azure.Identity

3.Add `[DefaultAzureCredential]`: https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/identity/Azure.Identity#defaultazurecredential

@jongio for notification.

### LINKED ISSUE #17670: Azure.Analytics.Synapse.Spark Readme issues
1.
Section [link:](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/synapse/Azure.Analytics.Synapse.Spark#install-the-package)
image
Suggestion:Add `[nuget]`: https://www.nuget.org/

2.
Section [link:](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/synapse/Azure.Analytics.Synapse.Spark#prerequisites)
image
Suggestion:

1.Add `[azure_sub]`: https://azure.microsoft.com/free/

2.Add hyperlink for `Azure Portal`: https://portal.azure.com/

3.Add `[azure_cli]`: https://docs.microsoft.com/cli/azure

3.
Section [link:](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/synapse/Azure.Analytics.Synapse.Spark#authenticate-the-client)
image
Suggestion:

1.Add `[spark_batch_client_class]`: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/synapse/Azure.Analytics.Synapse.Spark/src/Customization/SparkBatchClient.cs

2.Add `[spark_session_client_class]`: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/synapse/Azure.Analytics.Synapse.Spark/src/Customization/SparkSessionClient.cs

3.Add `[azure_identity]`: https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/identity/Azure.Identity

4.Add `[DefaultAzureCredential]`: https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/identity/Azure.Identity#defaultazurecredential

4.
Section [link:](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/synapse/Azure.Analytics.Synapse.Spark#spark-batch-job-examples)
image

Suggestion: Update  `Delete spark batch job` to  `Cancel spark batch job`
and update link to https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/synapse/Azure.Analytics.Synapse.Spark#cancel-spark-batch-job

@jongio for notification.

### PR DESCRIPTION
**Pull request by chamons**
- Fixes https://github.com/Azure/azure-sdk-for-net/issues/17666
- Fixes https://github.com/Azure/azure-sdk-for-net/issues/17667
- Fixes https://github.com/Azure/azure-sdk-for-net/issues/17668
- Fixes https://github.com/Azure/azure-sdk-for-net/issues/17669
- Fixes https://github.com/Azure/azure-sdk-for-net/issues/17670

### CODE REVIEW

**Comment by chamons** [comment]
FYI @Tianyun-G who helpfully found these.

File: sdk/synapse/Azure.Analytics.Synapse.Spark/README.md [hunk 1]
```diff
@@ -15,15 +15,15 @@ The complete Microsoft Azure SDK can be downloaded from the [Microsoft Azure Dow
...
+Install the Spark client library for Azure Synapse Analytics for .NET with [NuGet](https://www.nuget.org/):
 
 ```PowerShell
 dotnet add package Azure.Analytics.Synapse.Spark --version 0.1.0-preview.1
 ```
 
 ### Prerequisites
-* An [Azure subscription][azure_sub].
-* An existing Azure Synapse workspace. If you need to create an Azure Synapse workspace, you can use the Azure Portal or [Azure CLI][azure_cli].
+* An [Azure subscription](https://azure.microsoft.com/free/).
```
Comment (jsquire) [inline sdk/synapse/Azure.Analytics.Synapse.Spark/README.md:25]: You may want to consider expanding this a bit, maybe something like:

```suggestion
- **Azure Subscription:**  To use Azure services, including Azure Synapse, you'll need a subscription.  If you do not have an existing Azure account, you may sign up for a [free trial](https://azure.microsoft.com/free) or use your [Visual Studio Subscription](https://visualstudio.microsoft.com/subscriptions/) benefits when you [create an account](https://account.windowsazure.com/Home/Index).
```

**Review by jsquire — APPROVED** [review]

File: sdk/synapse/Azure.Analytics.Synapse.AccessControl/README.md [hunk 2]
```diff
@@ -13,15 +13,15 @@ The complete Microsoft Azure SDK can be downloaded from the [Microsoft Azure Dow
 For the best development experience, developers should use the official Microsoft NuGet packages for libraries. NuGet packages are regularly updated with new functionality and hotfixes.
 
 ### Install the package
-Install the Azure Synapse Analytics access control client library for .NET with [NuGet][nuget]:
+Install the Azure Synapse Analytics access control client library for .NET with [NuGet](https://www.nuget.org/):
```
Comment (Mohit-Chakraborty) [inline sdk/synapse/Azure.Analytics.Synapse.AccessControl/README.md:16]: Consider having more targeted links for Nuget (where it makes sense) so clicking on them lands at/near the desired package. Either a search query or a direct link to the package page.

File: sdk/synapse/Azure.Analytics.Synapse.Spark/README.md (hunk 1 above)
Comment (chamons) [reply sdk/synapse/Azure.Analytics.Synapse.Spark/README.md:25]: I like that. Let me add it 5x.

**Review by Mohit-Chakraborty — APPROVED** [review]

**Pull request merged**